src/tattoo.ts

```diff
@@ -239,16 +280,28 @@ function standardProgressBar(label: string, total: number) {
 }
 
 /**
- * Creates a branch with the given name on the given repo.
+ * Checks out a branch with a given name on a repo.
  *
  * returns a promise of the nodegit Branch object for the new branch.
  */
-async function checkoutNewBranch(
-    repo: nodegit.Repository, branchName: string): Promise<void> {
-  const commit = await repo.getHeadCommit();
-  const branch =
-      await nodegit.Branch.create(repo, branchName, commit, false);
-  return repo.checkoutBranch(branch);
+async function checkoutBranch(
+    repo: nodegit.Repository, branchName: string): Promise<nodegit.Repository> {
+  try {
```

Comment:
Given that you directly control whether there's an exception here, via the reject() call, you could just never call reject(), but in the error case log a warning and always resolve the repo. That would be a bit simpler.